celer/tests/test_lasso.py

```diff
@@ -210,7 +211,16 @@ def test_weights_lasso():
 
     assert_allclose(clf1.coef_, clf2.coef_ / weights)
 
+    # support inf weights
+    weights = np.ones(X.shape[1])
+    weights[0] = np.inf
+    alpha_max = norm(X.T.dot(y) / weights, ord=np.inf) / X.shape[0]
+
+    clf1 = Lasso(alpha=alpha_max, weights=weights,
```

Comment:
This adds a ``ConvergenceWarning``.
However, it is expected as we don't give enough iterations to the solver to converge (``max_iter=1``)

Reply:
Is there a reason no to use a bigger number of iteration ? Why is there a convergence warning whereas we fit at alpha_max, hence the solver starts at the solution (0)

Reply:
There is no reason not to put a big ``max_iter``. Indeed, I followed [your script](https://github.com/mathurinm/celer/pull/232#issue-1205467502) in PR #232.

Fitting with ``alpha_max`` doesn't mean that the solver will converge in 1 iteration.

Reply:
for alpha>=alpha_max the Lasso solution is 0. 0 is the initialization of the algorithm. So the solver converge in 0 iteration, because it starts at the solution.

Reply:
I checked the case of ``alpha >= alpha_max+epsilon`` and ``max_iter=1``. The solver doesn't converge at the first iteration.

Script to reproduce
```python
np.random.seed(1)
n_samples, n_features = 50, 100
X = np.random.randn(n_samples, n_features)
y = np.random.randn(n_samples)

weights = np.ones(n_features)
weights[0] = np.inf
alpha_max = norm(X.T @ y / weights, ord=np.inf) / n_samples

epsilon = 1e-5 * alpha_max
reg = Lasso(alpha=alpha_max+epsilon, weights=weights, max_iter=1)
reg.fit(X, y)
```

Output
```bash
c:\users\hp\desktop\celer-repo\celer\celer\homotopy.py:297: ConvergenceWarning: Objective did not converge: duality gap: 0.00011720669048692622, tolerance: 7.358563743764535e-05. Increasing `tol` may make the solver faster without affecting the results much.
Fitting data with very small alpha causes precision issues.
  sol = celer(
```